index.js

```diff
@@ -163,11 +172,11 @@ function launchSpecFromConnectionInfo(
     spawnOptions
   );
 
-  const runningKernel = child_process.spawn(
-    argv[0],
-    argv.slice(1),
-    fullSpawnOptions
-  );
+  const runningKernel = execa(argv[0], argv.slice(1), fullSpawnOptions);
+
+  runningKernel.on("exit", (code, signal) => cleanup(connectionFile));
+  runningKernel.on("error", (code, signal) => cleanup(connectionFile));
```

Comment:
Excellent! No more having to clean up that file on the nteract and hydrogen side. 🙏